mm.c

```diff
@@ -52,13 +52,25 @@ static char *main_free_list[MAX_POWER + 1];
 
 static char *heap_ptr;
 
+// Function Declarations
+static size_t find_free_list_index(size_t words);
+
 /*
 	Find the index of the free list which given size belongs to.
 	Returns index.
 	Index can be from 0 to MAX_POWER.
 */
 static size_t find_free_list_index(size_t words) {
+    unsigned int bytes = words * WORD_SIZE;
+    int index = 0;
+
+    while (index <= MAX_POWER && (bytes > 1))
```

Comment:
but (bytes > 1) has to be checked on every iteration, we left-shifting bytes by 1 until bytes = 0